connection/src/test/java/com/netcracker/ConnectionTest.java

```diff
@@ -0,0 +1,112 @@
+package com.netcracker;
+
+import com.netcracker.domain.Message;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.*;
+import java.lang.reflect.InvocationTargetException;
+import java.net.Socket;
+import java.net.SocketAddress;
+
+import static com.netcracker.domain.MessageType.TEXT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ConnectionTest {
```

Comment:
&#x1F34E; **Issue**
These tests are way to complex. Unit tests should always be simple to read. It would take another developer quite some time to understand what is going on. It won't do. You should either significantly simplify them or change the underlying implementation so that it would be easy to test it.